### ISSUE TITLE
a strange error while running the code

### ISSUE THREAD

**Issue by Lizhiweitracy** [jzbontar/mc-cnn#6]
Hello dear  Jure,

I am a newer in cnn. After looking your parper, I am trying to running the code following your instruction in web  .However I got a error when I ran the command  " ./main.lua kitti fast -a predict -net_fname net/net_kitti_fast_-a_train_all.t7 -left samples/input/kittiL.png -right samples/input/kittiR.png -disp_max 70 ".
the result is:
$ ./main.lua kitti fast -a predict -net_fname net/net_kitti_fast_-a_train_all.t7 -left samples/input/kittiL.png -right samples/input/kittiR.png -disp_max 70 -sm_terminate cnn
kitti fast -a predict -net_fname net/net_kitti_fast_-a_train_all.t7 -left samples/input/kittiL.png -right samples/input/kittiR.png -disp_max 70 -sm_terminate cnn 
Writing right.bin, 1 x 70 x 370 x 1226
Writing left.bin, 1 x 70 x 370 x 1226
luajit: ./main.lua:1052: bad argument #1 to 'resizeAs' (torch.CudaTensor expected, got torch.CudaLongTensor)
stack traceback:
	[C]: in function 'resizeAs'
	./main.lua:1052: in function 'stereo_predict'
	./main.lua:1098: in main chunk
	[C]: at 0x004064f0

as it shows ,it can write the right.bin and the left.bin,but coudn't write the disp.bin.........
looking forword to your reply  soon.

thank you very much

**Comment by jzbontar** [comment]
This was fixed in a recent commit (cd3a8fc). Can you please do a `git pull' and see if it works. Thanks!

**Comment by Lizhiweitracy** [comment]
yes !!!   It does work. thank you very much.  So happy

**Comment by Lizhiweitracy** [comment]
Hello dear Jure,
Today , i tried the command in the web including training ,compute the error rate ,prepare files for submission to the KITTI and so on . They all work well except the last command about ‘ Measure the runtime on a particular data set“。
The question looks strange and i don't know how to deal with it:

$ ./main.lua kitti fast -a time -sm_terminate cnn
kitti fast -a time -sm_terminate cnn 
luajit: ./main.lua:655: attempt to index global 'nnz' (a nil value)
stack traceback:
	./main.lua:655: in main chunk
	[C]: at 0x004064f0

thank you very much. Looking forword to your reply soon.It is so kind of you .

**Comment by jzbontar** [comment]
Should be fixed now. Let me know if it works.

**Comment by Lizhiweitracy** [comment]
hello dear  Jure,
thank you for your timely reply. I wonder the question which i referred to yesterday happened is because i interrupted the training step not so patient to wait for it.
 I just downloaded the trained networks and the the KITTI 2012 data set, then run the code "$ ./main.lua kitti fast -a time -sm_terminate cnn"  to  measure the runtime . Is this the reason for this question?
If i trained the networks by myself,  do you think this would be solved?
BTW: I am curious about :how long did you spend on training the ‘fast' and 'slow' networks respectively on the kitti  dataset?
Thank you .Any comments from you are appreciated!

**Comment by Lizhiweitracy** [comment]
hello dear Jure,
Sorry for misunderstanding the meaning last Friday. This morning, i follow the time command and running the new code ,the codes run well. No errors anymore .
Thanks .

**Comment by Lizhiweitracy** [comment]
Hi ,dear jure,
When i am reading the 'mc-cnn' code , i cann't understand the codes on line 1094 & line 1095 in main.lua:
 x0:add(-x0:mean()):div(x0:std())
  x1:add(-x1:mean()):div(x1:std())
Can you tell me what's the meaning of 'mean()' & 'std()'  ?
I am so sorry i can't find the definition of them.    THX a lot

**Comment by jzbontar** [comment]
You can find the definitions of mean() and std() in [torch's documentation](https://github.com/torch/torch7/blob/master/doc/maths.md).

**Comment by Lizhiweitracy** [comment]
Thank you very much

**Comment by Lizhiweitracy** [comment]
Hi ,dear jure,
After i understood layers connection of the accurate architecture ,i find the accurate architecture is too large to implement on my hardware. So i am learning the fast architecture, but i can't understand how the Normalize layer is implement. Could you please tell me what's the step to implement the Normalize layer?
I can't find it for a long time.
Thank you very much 
looking forword to your reply soon.

**Comment by jzbontar** [comment]
The Normalization layer takes as input a vector `x` and outputs `x / norm(x)`, where `norm(x)` is the L2 norm of vector `x` and `/` denotes element-wise division. The function that implements this operation can be found [`here`](https://github.com/jzbontar/mc-cnn/blob/master/adcensus.cu#L1310).

**Comment by Lizhiweitracy** [comment]
Got it . Thank you very much .best wishes for you .
